### DISCUSSION TITLE
Requesting Help: Cloudflare SSL Origin Cert | Not working with Wiki.js?

### DISCUSSION THREAD

**Discussion by GameServersHub** [requarks/wiki#6852] [Help / Questions] [answered]
Hello, I'd like to see if anyone has successfully installed Wiki.js on their Linux machines and supports Cloudflare (Full) SSL enabled with Origin Certification installed locally on the machine and was able to use Wiki.js to support it. Currently, I installed wiki.js by following the steps listed in https://docs.requarks.io/install/ubuntu and I can connect to my webpage by `HTTP` but I can't for the life of me get `HTTPS` to work with Cloudflare ssl full origin cert can anyone give me some advice and pointers on what I am doing incorrect or wrong? I would much rather use my origin cert than let's encrypt. Any solutions?

![image](https://github.com/requarks/wiki/assets/87217476/93368e50-99fc-4992-8857-caf7af35e57b)
![image](https://github.com/requarks/wiki/assets/87217476/f23eaed1-5eb1-4f09-a39d-46e87988327b)
![image](https://github.com/requarks/wiki/assets/87217476/8b3dc9c8-f809-44f6-99c0-0aa9a8cc78e4)
![image](https://github.com/requarks/wiki/assets/87217476/632ce7ff-91fb-4ac0-ad22-56b189a6f587)
![image](https://github.com/requarks/wiki/assets/87217476/7aa3e75c-3ed7-4941-b231-864577051578)
![image](https://github.com/requarks/wiki/assets/87217476/f5ac6b10-dfe6-498d-99ac-83a73bfcd249)

**Notes**:
- Since I'm running `Nginx Proxy Manager` I changed the default wiki.js docker ports from `80:3000 / 443:3443` to `8080:3000 / 4443:3443`
- All my ports are opened on the firewall.

**Comment by NGPixel** [comment, marked as answer]
Your config doesn't make any sense.

If you're using nginx-proxy, then you shouldn't be using SSL on Wiki.js at all.
Cloudflare -> 443 on nginx-proxy, configured with an origin SSL cert from Cloudflare -> 3000 (or whatever port you expose 3000 to locally)

If you're not using nginx-proxy, then you need to mount a custom config.yml file along with the cloudflare origin cert inside the docker container. The docker env vars only allow for a let's encrypt setup. See https://docs.requarks.io/install/config#custom-certificate on how to use a custom cert.

Using nginx-proxy is the far easiest solution though.
